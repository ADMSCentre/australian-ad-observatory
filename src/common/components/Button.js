import React from 'react';
import PropTypes from 'prop-types';
import BButton from 'react-bootstrap/Button';

const openPage = to => {
  const base = 'webpage/index.html';
  const url = to ? `${base}#${to}` : base;
  const extensionUrl = chrome.runtime.getURL(url);

  /* istanbul ignore next */
  return () => {
    window.open(extensionUrl, '_blank');
  };
};

const Button = ({ type, label, to, variant, onClick }) => {
  const doClick = onClick || openPage(to);

  return type === 'link' ? (
    <a href="#" onClick={doClick}>
      {label}
    </a>
  ) : (
    <BButton variant={variant} block onClick={doClick}>
      {label}
    </BButton>
  );
};
Button.displayName = 'Button';

Button.propTypes = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  type: PropTypes.oneOf(['link', 'button']),
  to: PropTypes.string, // you must have either a 'to' or 'onClick'
  onClick: PropTypes.func,
  variant: PropTypes.oneOf([
    'primary',
    'secondary',
    'success',
    'warning',
    'danger',
    'info',
    'light',
    'dark',
    'link'
  ])
};

Button.defaultProps = {
  type: 'button',
  to: null,
  onClick: null,
  variant: undefined
};

export default Button;
