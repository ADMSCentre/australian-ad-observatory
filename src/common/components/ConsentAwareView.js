import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import useConsent from 'common/hooks/useConsent';
import proceed from 'common/utils/proceed';

const ConsentAwareView = ({ children, whenGrantedIs, whenGrantedIsNot }) => {
  const { granted } = useConsent();

  return proceed(granted, whenGrantedIs, whenGrantedIsNot) ? <Fragment>{children}</Fragment> : null;
};
ConsentAwareView.displayName = 'Consent aware view';

ConsentAwareView.propTypes = {
  children: PropTypes.node.isRequired,
  whenGrantedIs: PropTypes.oneOf(['none', 'old', 'current']),
  whenGrantedIsNot: PropTypes.oneOf(['none', 'old', 'current'])
};

ConsentAwareView.defaultProps = {
  whenGrantedIs: null,
  whenGrantedIsNot: null
};

export default ConsentAwareView;
