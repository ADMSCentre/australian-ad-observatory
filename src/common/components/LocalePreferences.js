import React, { Fragment } from 'react';
import PreferenceSelector from './PreferenceSelector';
import { PREF_COUNTRY, PREF_LANGUAGE, PREF_AGE, PREF_GENDER } from 'common/keys';
import { COUNTRIES } from 'common/countries';
import { LANGUAGES } from 'common/languages';
import { AGES } from 'common/ages';
import { GENDERS } from 'common/genders';

const toArray = object => Object.keys(object).map(value => ({ value, text: object[value] }));

const LocalePreferences = () => (
  <Fragment>
    <PreferenceSelector
      options={toArray(COUNTRIES)}
      label="Your Country"
      storageKey={PREF_COUNTRY}
    />
    <PreferenceSelector
      options={toArray(LANGUAGES)}
      label="Your Language"
      storageKey={PREF_LANGUAGE}
    />
    <PreferenceSelector options={toArray(AGES)} label="Age" storageKey={PREF_AGE} />
    <PreferenceSelector options={toArray(GENDERS)} label="Gender" storageKey={PREF_GENDER} />
  </Fragment>
);
LocalePreferences.displayName = 'Locale preferences';

export default LocalePreferences;
