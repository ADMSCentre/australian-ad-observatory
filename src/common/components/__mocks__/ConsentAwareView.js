/* eslint-disable react/prop-types */
import React, { Fragment } from 'react';
import proceed from 'common/utils/proceed';

let granted;

export const setGranted = gtd => {
  granted = gtd;
};

const ConsentAwareView = ({ children, whenGrantedIs, whenGrantedIsNot }) =>
  proceed(granted, whenGrantedIs, whenGrantedIsNot) ? <Fragment>{children}</Fragment> : null;

export default ConsentAwareView;
