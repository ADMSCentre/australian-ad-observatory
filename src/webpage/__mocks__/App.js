import React from 'react';

const App = () => <div name="app" />;

export default App;
