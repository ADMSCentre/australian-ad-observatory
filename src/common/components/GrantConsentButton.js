import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

import useConsent from 'common/hooks/useConsent';
import useNavigation from 'common/hooks/useNavigation';

const GrantConsentButton = ({ label, to }) => {
  const { granted, saveConsent } = useConsent();
  const { goto } = useNavigation();

  const saveAndGo = () => {
    saveConsent();
    if (to) goto(to);
  };

  return granted === null || granted === 'current' ? null : (
    <Button variant="primary" onClick={saveAndGo} label={label} />
  );
};
GrantConsentButton.displayName = 'Grant aware button';

GrantConsentButton.propTypes = {
  label: PropTypes.string.isRequired,
  to: PropTypes.string
};

GrantConsentButton.defaultProps = {
  to: null
};

export default GrantConsentButton;
