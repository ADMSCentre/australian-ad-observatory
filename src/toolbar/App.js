import React, { Fragment } from 'react';
import Container from 'react-bootstrap/Container';

import ConsentAwareView from 'common/components/ConsentAwareView';
import Button from 'common/components/Button';
import MainMenu from './components/MainMenu';

import 'bootstrap/dist/css/bootstrap.min.css';
import './App.less';

const App = () => (
  <Fragment>
    <ConsentAwareView whenGrantedIs="none">
      <Container>
        <h3>{process.env.TITLE}</h3>
        <div className="mb-1">
          <p>To enable the extension, please consent to our collection of your data.</p>
          <Button label="Review Consent" to="terms" />
        </div>
      </Container>
    </ConsentAwareView>
    <ConsentAwareView whenGrantedIs="old">
      <Container>
        <div className="mb-1">
          <h3>{process.env.TITLE}</h3>
          <p>
            To enable the extension, please consent to our collection of some more of your data.
          </p>
          <Button label="Review Consent" to="terms" />
        </div>
      </Container>
    </ConsentAwareView>
    <ConsentAwareView whenGrantedIs="current">
      <MainMenu />
    </ConsentAwareView>
  </Fragment>
);
App.displayName = 'App';

export default App;
