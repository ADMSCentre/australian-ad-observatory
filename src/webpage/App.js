import React, { Fragment } from 'react';
import Container from 'react-bootstrap/Container';

// import useArchive from 'common/hooks/useArchive';
import useNavigation from 'common/hooks/useNavigation';

import ConsentAwareView from 'common/components/ConsentAwareView';

import Header from './components/Header';
import Navigation from './components/Navigation';

// import NewsPage from './pages/NewsPage';
import AboutPage from './pages/AboutPage';
import ArchivePage from './pages/ArchivePage';
import PreferencesPage from './pages/PreferencesPage';
// import SharePage from './pages/SharePage';
import PrivacyPage from './pages/PrivacyPage';
import TermsPage from './pages/TermsPage';
import HelpPage from './pages/HelpPage';

import 'bootstrap/dist/css/bootstrap.min.css';
import './App.less';

const routes = {
  // news: NewsPage,
  about: AboutPage,
  archive: ArchivePage,
  preferences: PreferencesPage,
  // share: SharePage,
  privacy: PrivacyPage,
  terms: TermsPage,
  help: HelpPage
};

const App = () => {
  const { current } = useNavigation();
  const Page = routes[current] || AboutPage;

  return (
    <Fragment>
      <Header />
      <div className="d-flex" id="wrapper">
        <ConsentAwareView whenGrantedIsNot="current">
          <div id="page-content-wrapper">
            <TermsPage />
          </div>
        </ConsentAwareView>
        <ConsentAwareView whenGrantedIs="current">
          <Navigation />
          <div id="page-content-wrapper">
            <Container fluid>
              <Page />
            </Container>
          </div>
        </ConsentAwareView>
      </div>
    </Fragment>
  );
};
App.displayName = 'App';

export default App;
