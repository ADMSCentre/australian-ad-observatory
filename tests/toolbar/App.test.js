import React from 'react';
import renderer from 'react-test-renderer';

import App from 'toolbar/App';

import { setGranted } from 'common/components/ConsentAwareView';

jest.mock('common/components/ConsentAwareView');
jest.mock('toolbar/components/MainMenu');

let tree;

['none', 'old', 'current'].forEach(granted => {
  describe(`when consent is ${granted}`, () => {
    beforeAll(() => {
      setGranted(granted);
      tree = renderer.create(<App />);
    });

    it('renders correctly', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
