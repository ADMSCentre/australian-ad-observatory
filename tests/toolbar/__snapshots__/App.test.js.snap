// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when consent is current renders correctly 1`] = `
<div>
  I am a Main Menu
</div>
`;

exports[`when consent is none renders correctly 1`] = `
<div
  className="container"
>
  <h3>
    Test Project
  </h3>
  <div
    className="mb-1"
  >
    <p>
      To enable the extension, please consent to our collection of your data.
    </p>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Review Consent
    </button>
  </div>
</div>
`;

exports[`when consent is old renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="mb-1"
  >
    <h3>
      Test Project
    </h3>
    <p>
      To enable the extension, please consent to our collection of some more of your data.
    </p>
    <button
      className="btn btn-primary btn-block"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Review Consent
    </button>
  </div>
</div>
`;
